File: application/app/src/main/java/com/yahyeet/boardbook/presenter/HomePresenter.java
package com.yahyeet.boardbook.presenter;

import android.content.Context;
import android.content.Intent;

import com.yahyeet.boardbook.activity.home.IHomeActivity;
import com.yahyeet.boardbook.activity.profile.ProfileActivity;
import com.yahyeet.boardbook.activity.account.AccountManagerActivity;

public class HomePresenter {

	private IHomeActivity homeActivity;

	public HomePresenter(IHomeActivity homeActivity) {
		this.homeActivity = homeActivity;
	}


	public void startLoggedInUserProfile(){
		Context context = (Context) homeActivity;
		Intent startProfile = new Intent( context, ProfileActivity.class);
		startProfile.putExtra("UserId", BoardbookSingleton.getInstance().getAuthHandler().getLoggedInUser().getId());
		context.startActivity(startProfile);
	}

	public void logOut(){
		BoardbookSingleton.getInstance().getAuthHandler().logout();
		Context context = (Context) homeActivity;
		Intent startLogin = new Intent( context, AccountManagerActivity.class);
		context.startActivity(startLogin);
	}
}

File: application/app/src/main/java/com/yahyeet/boardbook/presenter/HomePresenter.kt
package com.yahyeet.boardbook.presenter

import android.content.Context
import android.content.Intent

import com.yahyeet.boardbook.activity.home.IHomeActivity
import com.yahyeet.boardbook.activity.profile.ProfileActivity
import com.yahyeet.boardbook.activity.account.AccountManagerActivity

class HomePresenter(private val homeActivity: IHomeActivity) {


    fun startLoggedInUserProfile() {
        val context = homeActivity as Context
        val startProfile = Intent(context, ProfileActivity::class.java)
        startProfile.putExtra("UserId", BoardbookSingleton.getInstance().authHandler.loggedInUser!!.id)
        context.startActivity(startProfile)
    }

    fun logOut() {
        BoardbookSingleton.getInstance().authHandler.logout()
        val context = homeActivity as Context
        val startLogin = Intent(context, AccountManagerActivity::class.java)
        context.startActivity(startLogin)
    }
}
